utility/migration/migration.ts

```diff
@@ -2,20 +2,20 @@ import compareVersions from 'compare-versions'
 import { ExperimentType } from '../../types/common'
 
 //TODO: Compare json to current ExperimentType and set missing fields to default values?
-export const migrate = (json: any): ExperimentType => {
+export const migrate = (json: any, stopAtVersion=MIGRATIONS[MIGRATIONS.length - 1].version): ExperimentType => {
   const version = json.info.dataFormatVersion !== undefined ? json.info.dataFormatVersion : "0"
   const firstMigration = MIGRATIONS.find(m => compareVersions(version, m.version) === -1)
-  return firstMigration === undefined ? json : doMigrations(firstMigration, json)
+  return firstMigration === undefined || compareVersions(firstMigration.version, stopAtVersion) > 0 ? json : doMigrations(firstMigration, json, stopAtVersion)
 }
 
-const doMigrations = (migration: Migration, json: any): any => {
+const doMigrations = (migration: Migration, json: any, stopAtVersion: string): any => {
   json = migration.converter(json)
   const migrationIndex = MIGRATIONS.findIndex(m => m === migration)
-  const isLastMigration = migrationIndex === MIGRATIONS.length - 1
+  const isLastMigration = migrationIndex === MIGRATIONS.length - 1 || migration.version === stopAtVersion
```

Comment:
Can _migrationIndex === MIGRATIONS.length - 1_ be removed now that there is a stopAtVersion with a default?

Reply:
Yes it could be removed, however, I think it would be better to keep it to prevent errors where a user enters a non-existing stopVersion